src/scripting/lua_widget_attributes.cpp

```diff
@@ -47,10 +52,17 @@
 #include <utility>
 #include <vector>
 
-
 static lg::log_domain log_scripting_lua("scripting/lua");
 #define ERR_LUA LOG_STREAM(err, log_scripting_lua)
 
+#define INVALIDATE_LAYOUT() \
```

Comment:
Everything looks good to me, but I would prefer if this macro were a function, or if it's easier as a macro, just do a one-liner, like so:
```cpp
#define INVALIDATE_LAYOUT() if(auto window = w.get_window()) { window->invalidate_layout(); }
```
(Unless there was a specific reason you needed it as a do-while loop?)